feat: reply trailers support (#3794)
fastify/fastify: docs/Reference/Reply.md

Returns a boolean indicating if the specified header has been set.

### .trailer(key, function)
<a id="trailer"></a>

> Sets a response trailer.

Reviewer: Please
document in which moments we call the functions as well as that the functions must be synchronous.

Returns a boolean indicating if the specified header has been set.

### .trailer(key, function)
<a id="trailer"></a>

Sets a response trailer. Trailer usually used when you want some header that require heavy resources to be sent after the `data`, for example `Server-Timing`, `Etag`. It can ensure the client get the response data as soon as possible.

*Note: The header `Transfer-Encoding: chunked` will be added once you use the trailer. It is a hard requipment for using trailer in Node.js.*

> *Note: Currently, the computation function only support synchronous function. That means `async-await` and `promise` is not supported.*

Reviewer 2: ```suggestion
*Note: Currently, the computation function only supports synchronous function. That means `async-await` and `promise` are not supported.*
```

```js
reply.trailer('server-timing', function() {
  return 'db;dur=53, app;dur=47.2'
})

const { createHash } = require('crypto')
// trailer function also recieve two argument
// @param {object} reply fastify reply
// @param {string|Buffer|null} payload payload that already sent, note that it will be null when stream is sent
reply.trailer('content-md5', function(reply, payload) {
  const hash = createHash('md5')

>   hash.update(payload);

Reviewer 2: ```suggestion
  hash.update(payload)
```

  return hash.disgest('hex')
})
```

> 
> 

Reviewer 2: ```suggestion
```
